From 554f7baed050f89ffc2a7192d3071e8c5420f6d3 Mon Sep 17 00:00:00 2001
From: Mikko Rapeli <mikko.rapeli@linaro.org>
Date: Fri, 25 Aug 2023 10:35:28 +0000
Subject: [PATCH] openssh regress/Makefile: print logs if test fails

Some tests are failing in CI runs and reproduction has failed. Print
the captured sshd and ssh client logs if test fails. This should
help to fix the root causes.

Reference: https://bugzilla.yoctoproject.org/show_bug.cgi?id=15178

Signed-off-by: Mikko Rapeli <mikko.rapeli@linaro.org>
---
 regress/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Upstream-Status: Submitted [https://github.com/openssh/openssh-portable/pull/437]

diff --git a/regress/Makefile b/regress/Makefile
index d80bf59..a972dff 100644
--- a/regress/Makefile
+++ b/regress/Makefile
@@ -229,7 +229,7 @@ t-exec:	${LTESTS:=.sh}
 		done; \
 		if [ "x$${skip}" = "xno" ]; then \
 			echo "run test $${TEST}" ... 1>&2; \
-			(env SUDO="${SUDO}" TEST_ENV=${TEST_ENV} ${TEST_SHELL} ${.CURDIR}/test-exec.sh ${.OBJDIR} ${.CURDIR}/$${TEST}) || exit $$?; \
+			(env SUDO="${SUDO}" TEST_ENV=${TEST_ENV} ${TEST_SHELL} ${.CURDIR}/test-exec.sh ${.OBJDIR} ${.CURDIR}/$${TEST}) || (echo return value: $$?; echo capturing logs; cat *.log; exit 1); \
 		else \
 			echo skip test $${TEST} 1>&2; \
 		fi; \
-- 
2.34.1

